src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -188,6 +189,7 @@ trait ContextErrors {
         // scala/bug#3971 unwrapping to the outermost Apply helps prevent confusion with the
         // error message point.
         def callee = {
+          @tailrec
```

Comment:
Can we make the `callee` into a `val` or a `lazy val`? It looks to be called twice in the `issueError`, which is called from the `match` below.